src/solvers/sat/dimacs_cnf.cpp

```diff
@@ -19,6 +19,11 @@ dimacs_cnft::dimacs_cnft():break_lines(false)
 {
 }
 
+dimacs_cnft::dimacs_cnft(message_handlert &message_handler) : dimacs_cnft()
+{
+  log.set_message_handler(message_handler);
```

Comment:
This is going to be a bit painful, but let's please try to use the constructor all the way, i.e., it would be `dimacs_cnft(message_handler)` above.

Reply:
That sounds like a good idea but this would go a bit beyond scope of a refactoring PR, and my initial statement "No functional changes." wouldn't be strictly true anymore.

Reply:
Well, if we have no bugs (surely true) then it wouldn't entail any functional changes, because each time we construct a `dimacs_cnft` we surely also initialized its message handler.

Reply:
It we leave it initialized, doesn't it just means that some messages wouldn't get printed? This would be difficult to detect. It's also a bit difficult to manually check, as a lot of classes inherit from `dimacs_cnft`...

Reply:
I think any introduced functional change is a strict improvement here. Any class deriving should be forced to call a constructor that takes a message handler.